src/openrct2/ride/VehiclePaint.cpp

```diff
@@ -3663,7 +3663,15 @@ void VehicleVisualDefault(
 {
     if (vehicle->Pitch < std::size(PaintFunctionsByPitch))
     {
-        PaintFunctionsByPitch[vehicle->Pitch](session, vehicle, imageDirection, z, carEntry);
+        if (vehicle->isReversed)
+        {
+            auto imagePitch = vehicle->GetInvertedPitch();
+            PaintFunctionsByPitch[imagePitch](session, vehicle, (imageDirection + 16) % 32, z, carEntry);
```

Comment:
Magic numbers 16 and 32, this should use OpenRCT2::Entity::Yaw values and functions.